plugins/headers.rst

```diff
@@ -0,0 +1,143 @@
+Header Plugins
+==============
+
+Header plugins are useful to manage request headers. Many operations are possible:
+
+Default headers values
+----------------------
+
+The plugin Http\Plugins\HeaderDefaultPlugin allows to set default values for given headers.
+That means if the request hasn't already a given header it will be set,
+but if the request already has a value for the given header then the request wont be changed
+
+.. code:: php
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Plugins\PluginClient;
+    use Http\Plugins\HeaderDefaultPlugin;
+
+    $defaultUserAgent = 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.1';
+
+    $headerDefaultPlugin = new HeaderDefaultPlugin([
+        'User-Agent' => $defaultUserAgent
+    ]);
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$headerDefaultPlugin]
+    );
+
+Mandatory headers values
+------------------------
+
+The Http\Plugins\HeaderSetPlugin allows to fix values of given header. That mean that any request passing through
+this plugin will have the same values for given headers
+
+.. code:: php
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Plugins\PluginClient;
+    use Http\Plugins\HeaderSetPlugin;
+
+    $userAgent = 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.1';
+
+    $headerSetPlugin = new HeaderSetPlugin([
+        'User-Agent' => $userAgent,
+        'Accept' => 'application/json'
+    ]);
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$headerSetPlugin]
+    );
+
+
+
+Removing headers
+----------------
+
+The Http\Plugins\HeaderRemovePlugin allows to remove given headers from the request.
+That mean that any request passing through this plugin will be freed of the given headers.
```

Comment:
Also the two sentences seem to be somewhat identical to me. I see that it matches the above two, but I think it would be better to somehow make them into one.